client/client.go

```diff
@@ -255,7 +255,7 @@ type ReadyCheckData struct {
 }
 
 // CheckReady application heath check using ManifestOutputData params
-func (m *K8sClient) CheckReady(namespace string, c *ReadyCheckData) error {
+func (m *K8sClient) CheckReady(namespace string, c *ReadyCheckData, appCount int) error {
```

Comment:
We now have 2 checks in `CheckReady`:
- Checking we are seeing all pods with `app` label
- Checking all pods are `Running` and that all containers are `Ready`
Both checks are doing almost the same thing
I think the race is here in `WaitPodsReady` between
```
podList, err := m.ListPods(ns, rcd.ReadinessProbeCheckSelector)
```
And the code that checks that list afterwards, can we, maybe, check that list for pods count, not apps count?

Reply:
So the problem is that we are not validating list for consistency with our deployment.
We are also checking all the apps with the label `app` are equal to all the pods we've found in replicas, I think it'd break if there will be a single deployment without `app` label? Maybe we can remove that convention now, when we started to check pods number in deployment directly?

Reply:
Maybe we can just check it as `helm` do
<img width="1203" alt="image" src="https://user-images.githubusercontent.com/14962379/214265839-b6b9b4c4-151e-469b-820d-b85e2cda8ccb.png">

Reply:
In our case `Desired - maxUnavailable` can do the trick, maybe?

Reply:
I think that will be a better direction. Will make a proper ticket to fix this a better way. For now we will continue with this to unblock people but we definitely need a better solution. I cleaned up the code a bit and broke it out into logical functions for now.

Reply:
https://github.com/helm/helm/blob/76157c6d06f981a42bfbb9ec636bae5443cb2c8f/pkg/kube/ready.go#L91